### DISCUSSION TITLE
"FILE" identifier undefined in C - MacOS 12

### DISCUSSION THREAD

**Discussion by gbartuccio** [microsoft/vscode-cpptools#8868] [Q&A]
Hi, I wanted to use the FILE structure which is included in stdio.h, but the "FILE* fp" doesn't get recognized, saying that the "FILE identifier is undefined". Im running VSCode on MacOS 12.2.1 M1, gcc installed via Homebrew.

#include <stdio.h>
#include <stdlib.h>
main(){
  char c;
  FILE* fp = fopen("prova.txt","w");
  if (fp==NULL) exit(1);
  else {
    while ((c=getchar())!=EOF)
      fputc(c,fp);
    fclose(fp);
  }
}

**Comment by sean-mcmanus** [comment]
Can you run C/C++: Log Diagnostics and check if the compilerPath and system includes are as expected?

**Reply by michelleangela** [reply]
Could you also run `gcc --version` from the terminal to see how the gxx-include-dir is configured the gcc compiler?

**Reply by gbartuccio** [reply]
> Can you run C/C++: Log Diagnostics and check if the compilerPath and system includes are as expected?

how do I do that?

**Reply by michelleangela** [reply]
Type the command `C/C++: Log Diagnostics` in the [VS Code command palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) and press enter to run the command.

**Reply by gbartuccio** [reply]
That's the output:


-------- Diagnostics - 2/16/2022, 7:50:49 PM
Version: 1.8.4
Current Configuration:
{
    "name": "Mac",
    "includePath": [
        "${workspaceFolder}/**",
        "/opt/homebrew/Cellar/gcc/11.2.0_1/include/c++/11.1.0",
        "/opt/homebrew/Cellar/gcc/11.2.0_1/include/c++/11.1.0/tr1",
        "/opt/homebrew/Cellar/gcc/11.2.0_1/include/c++/11.1.0/aarch64-apple-darwin20"
    ],
    "defines": [],
    "macFrameworkPath": [],
    "compilerPath": "/opt/homebrew/bin/gcc",
    "cStandard": "gnu17",
    "cppStandard": "gnu++17",
    "intelliSenseMode": "macos-gcc-arm64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true
}
Translation Unit Mappings:
[ /Users/gb/Documents/program.c ]:
    /Users/gb/Documents/program.c
Translation Unit Configurations:
[ /Users/gb/Documents/program.c ]:
    Process ID: 4917
    Memory Usage: 0 MB
    Compiler Path: /opt/homebrew/bin/gcc
    Includes:
        /opt/homebrew/Cellar/gcc/11.2.0_1/include/c++/11.1.0
        /opt/homebrew/Cellar/gcc/11.2.0_1/include/c++/11.1.0/tr1
        /opt/homebrew/Cellar/gcc/11.2.0_1/include/c++/11.1.0/aarch64-apple-darwin20
        /opt/homebrew/Cellar/gcc/11.2.0_1/lib/gcc/11/gcc/aarch64-apple-darwin20/11.1.0/include
        /opt/homebrew/Cellar/gcc/11.2.0_1/lib/gcc/11/gcc/aarch64-apple-darwin20/11.1.0/include-fixed
        /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include
        /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/System/Library/Frameworks
    Standard Version: c17
    IntelliSense Mode: macos-gcc-arm64
    Other Flags:
        --gcc
        --gnu_version=110100
Total Memory Usage: 0 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 8705
Number of files parsed: 1058

**Comment by michelleangela** [comment]
@gbartuccio 
The log diagnostic shows the memory usage of the IntelliSense is 0 MB, which indicates that it probably crashed.

Could you also get logs of the language server by following the instructions at https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server? These logs may show if IntelliSense is crashing or when it fails to work.

**Comment by gbartuccio** [comment]
I quickly fixed by just updating gcc via terminal, I guess something just went wrong during the update from MacOS 11 to 12.
